azkaban-common/src/main/java/azkaban/server/SlaRequestUtils.java

```diff
@@ -85,8 +93,42 @@ public static List<SlaOption> parseSlaOptions(final HttpServletRequest req, fina
     return slaOptions;
   }
 
-  private static SlaOption parseSlaSetting(final String set, final String flowName, final List<String> emails,
-      final Map<String, Map<String, String>> alertersConfigs) throws ServletException {
+  private static Map<String, Map<String, List<String>>> getStringListMapFromStringMap(
+      final Map<String, Map<String, String>> alertersConfsFromReqParams) {
+    final Map<String, Map<String, List<String>>> result = new HashMap<>();
+    for (final String alerter : alertersConfsFromReqParams.keySet()) {
+      final Map<String, String> alerterConfigs = alertersConfsFromReqParams.get(alerter);
+      final Map<String, List<String>> processedConfigs = new HashMap<>();
+      for (final String propKey : alerterConfigs.keySet()) {
+        processedConfigs.put(propKey, getListFromString(alerterConfigs.get(propKey),
+            alerter.equals(Emailer.ALERTER_NAME)));
+      }
+      result.put(alerter, processedConfigs);
+    }
+    return result;
+  }
+
+  private static List<String> getListFromString(final String str, final boolean spacesAreDelims) {
+    if (str == null || str.trim().isEmpty()) {
+      return Collections.emptyList();
+    }
+    String delimRegex = "\\s*,\\s*|\\s*;\\s*";
+    if (spacesAreDelims) {
+      delimRegex = "\\s*,\\s*|\\s*;\\s*|\\s+";
+    }
+    return Arrays.asList(str.trim().split(delimRegex));
+  }
+
+  public static Set<String> getSetFromString(final String val) {
+    if (val == null || val.trim().length() == 0) {
+      return Collections.emptySet();
+    }
+    return new HashSet<>(Arrays.asList(val.split("\\s*,\\s*")));
```

Comment:
Can we abstract "\\s*,\\s*" ? Seems like magic to me...

Reply:
Deleted the method, it shouldn't be here as explained above.